framework/src/main/java/org/checkerframework/framework/type/GenericAnnotatedTypeFactory.java

```diff
@@ -268,6 +269,15 @@
                     IdentityHashMap<Node, TransferResult<Value, Store>>>
             flowResultAnalysisCaches;
 
+    /**
+     * Subcheckers share the same ControlFlowGraph for each analyzed code statement. This maps from
+     * code statements to the shared control flow graphs. This map is null in all subcheckers (i.e.
+     * it is null exactly when {@code this.checker.parentChecker} is non-null, and vice-versa).
+     *
+     * <p>The initial capacity of the map is set by {@link #getCacheSize()}.
+     */
+    protected @Nullable Map<Tree, ControlFlowGraph> subcheckerSharedCFG;
```

Comment:
I am concerned that this map is not reset by `setRoot`.  It seems to me that it will grow without bound and will prevent the garbage collection of its contents.  Both of these are likely to lead to memory problems.

Reply:
Your concern about memory leakage is reasonable.

Right now, this map is set when the first subchecker is run, in the ultimate parent checker (so that it can be shared by e.g. sibling checkers of the first checker to run). (Note that this map is always `null` in subcheckers that have a non-null parent.) If the map were to be reset by `setRoot`, then when that ultimate parent checker starts running (and `setRoot` is called on its ATF), the shared CFG would be cleared - defeating the point.

Ideally, this map should be cleared whenever:
* if the checker has no subcheckers, when `setRoot` is called (it would be easy to implement this case)
* if the checker does have subcheckers, when `setRoot` is called on *only the first* subchecker to run (i.e. the subchecker that is responsible for creating the CFG). This is the tricky case - I'm not sure how to determine in a given call to `setRoot` that the checker whose root is being set is the first subchecker to run (is the run order even defined, or is it arbitrary which subchecker runs first?).

It might be possible to add a flag (call it `okayToOverwriteCFG` inside the ultimate parent that is set to true whenever `setRoot` is called on that checker's ATF (i.e. when it itself is running, meaning all subcheckers have completed), and set it to false whenever the map is re-initialized. I need to think that through a bit more though to make sure it will work, and if you have better ideas I'd love to hear them.

Reply:
What's the status of this comment?

Reply:
This comment wasn't marked as resolved.  Did you make the changes you propose in this comment?  I think you did make them, but I wasn't sure.  (`subcheckerSharedCFG` is cleared in setRoot via a call to`clearSharedCFG`.)

Reply:
Sorry, I got confused by Github's interface before, and didn't realize that this comment chain is what's being referred to.

To answer your question, yes, I did make these changes. This comment should have been marked resolved.